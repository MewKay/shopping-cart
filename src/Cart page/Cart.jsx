import { Link, useOutletContext } from "react-router-dom";
import CartItem from "./CartItem";
import styles from "./Cart.module.css";

const Cart = () => {
  const { orderSubtotal, itemNumber, cart } = useOutletContext();
  const isCartEmpty = cart.length <= 0;
  const itemPlural = itemNumber > 1 && "s";
  const deliveryFee = 0.0;
  const discount = 0.0;
  const orderTotal = orderSubtotal + deliveryFee + discount;

  return (
    <main className={styles["main-container"]}>
      <div className={styles["container"]}>
        {isCartEmpty ? (
          <p className={styles["empty-message"]}>Your Cart is empty</p>
        ) : (
          <>
            <div className={styles["cart-container"]}>
              <header className={styles["cart-header"]}>
                <h2>Your Shopping Cart</h2>
                <p>
                  {itemNumber} different item{itemPlural}
                </p>
              </header>
              <div className={styles["cart-list"]}>
                <p className={styles["cart-list-header"]}>PRODUCT DETAILS</p>
                <p className={styles["cart-list-header"]}>QUANTITY</p>
                <p className={styles["cart-list-header"]}>PRICE</p>
                <p className={styles["cart-list-header"]}>TOTAL</p>
                <p></p>
                {cart.map((item, index) => (
                  <CartItem
                    key={item.productDetails.id}
                    cartItem={item}
                    cartItemIndex={index}
                  />
                ))}
              </div>
              <Link to={"/store/all"} className={styles["go-shop-link"]}>
                {"<-- Continue Shopping"}
              </Link>
            </div>
            <div className={styles["summary-container"]}>
              <h3 className={styles["summary-title"]}>Order Summary</h3>
              <div className={styles["summary-info"]}>
                <div>
                  <p className={styles["summary-labels"]}>SUBTOTAL</p>
                  <p className={styles["summary-prices"]}>${orderSubtotal}</p>
                </div>
                <div>
                  <p className={styles["summary-labels"]}>DELIVERY</p>
                  <p className={styles["summary-prices"]}>${deliveryFee}</p>
                </div>
                <div>
                  <p className={styles["summary-labels"]}>DISCOUNT</p>
                  <p className={styles["summary-prices"]}>${discount}</p>
                </div>
                <div className={styles["total-container"]}>
                  <p className={styles["summary-labels"]}>TOTAL</p>
                  <p className={styles["summary-prices"]}>${orderTotal}</p>
                </div>
                <button className={styles["checkout-button"]}>CHECKOUT</button>
              </div>
            </div>
          </>
        )}
      </div>
    </main>
  );
};

export default Cart;
