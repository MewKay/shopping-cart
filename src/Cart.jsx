const Cart = () => {
  return (
    <div>
      <div>
        <header>
          <h2>Your Shopping Cart</h2>
        </header>
        <ul>
          <li>Product Details</li>
          <li>Quantity</li>
          <li>Price</li>
          <li>Total</li>
        </ul>
      </div>
    </div>
  );
};

export default Cart;
