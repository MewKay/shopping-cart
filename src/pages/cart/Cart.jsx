import { useOutletContext } from "react-router-dom";
import styles from "./Cart.module.css";
import CartList from "./cart-list/CartList";

const Cart = () => {
  const { orderSubtotal, cart } = useOutletContext();
  const isCartEmpty = cart.length <= 0;
  const deliveryFee = 0.0;
  const discount = 0.0;
  const orderTotal = orderSubtotal + deliveryFee + discount;

  return (
    <main className={styles["main-container"]}>
      <div className={styles["container"]}>
        {isCartEmpty ? (
          <p className={styles["empty-message"]}>Your Cart is empty</p>
        ) : (
          <>
            <CartList />
            <div className={styles["summary-container"]}>
              <h3 className={styles["summary-title"]}>Order Summary</h3>
              <div className={styles["summary-info"]}>
                <div>
                  <p className={styles["summary-labels"]}>SUBTOTAL</p>
                  <p className={styles["summary-prices"]}>${orderSubtotal}</p>
                </div>
                <div>
                  <p className={styles["summary-labels"]}>DELIVERY</p>
                  <p className={styles["summary-prices"]}>${deliveryFee}</p>
                </div>
                <div>
                  <p className={styles["summary-labels"]}>DISCOUNT</p>
                  <p className={styles["summary-prices"]}>${discount}</p>
                </div>
                <div className={styles["total-container"]}>
                  <p className={styles["summary-labels"]}>TOTAL</p>
                  <p className={styles["summary-prices"]}>${orderTotal}</p>
                </div>
                <button className={styles["checkout-button"]}>CHECKOUT</button>
              </div>
            </div>
          </>
        )}
      </div>
    </main>
  );
};

export default Cart;
