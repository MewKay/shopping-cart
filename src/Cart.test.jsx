import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Cart from "./Cart";

describe("Cart component", () => {
  it("renders correctly", () => {
    const { container } = render(<Cart />);

    expect(container).toMatchSnapshot();
  });
});
