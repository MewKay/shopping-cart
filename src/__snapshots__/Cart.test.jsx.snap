// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart component > renders correctly 1`] = `
<div>
  <div>
    <div>
      <header>
        <h2>
          Your Shopping Cart
        </h2>
      </header>
      <ul>
        <li>
          Product Details
        </li>
        <li>
          Quantity
        </li>
        <li>
          Price
        </li>
        <li>
          Total
        </li>
      </ul>
    </div>
  </div>
</div>
`;
